### PR TITLE
Update build for gfsv17 test to use -D32BIT=ON and set fewer tests for S4

### PR DESCRIPTION
**Pull request by JessicaMeixner-NOAA**
## Description
Move cpld_control_gfsv17 to -D32BIT=ON build section as GFSv17 will use this option.  Description of the evaluation by @LydiaStefanova-NOAA can be found [here](https://docs.google.com/presentation/d/1wHdDBkxLT8sPM0ClYeLLo3YXulOLittXm4EqCkgOaTo/edit?usp=sharing) comparing -D32BIT=ON vs OFF for HR1 runss. 

### Input data additions/changes
- [x] No changes are expected to input data.
- [ ] There will be new input data. 
- [ ] Input data will be updated. 

### Anticipated changes to regression tests:
- [ ] No changes are expected to any regression test. 
- [x] Changes are expected to the following tests: cpld_control_gfsv17 

## Subcomponents involved:
- [ ] AQM
- [ ] CDEPS
- [ ] CICE
- [ ] CMEPS
- [ ] CMakeModules
- [ ] FV3
- [ ] GOCART
- [ ] HYCOM
- [ ] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [x] none

### Combined with PR's (If Applicable):
- combined in https://github.com/ufs-community/ufs-weather-model/pull/1701

## Commit Queue Checklist:

- [x] Link PR's from all sub-components involved
- [x] Confirm reviews completed in sub-component PR's
- [x] Add all appropriate labels to this PR.
- [x] Run full RT suite on either Hera/Cheyenne with both Intel/GNU compilers
- [x] Add list of any failed regression tests to "Anticipated changes to regression tests" section.

## Testing Day Checklist:

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.
- [ ] Move new/updated input data on RDHPCS Hera and propagate input data changes to all supported systems.

### Testing Log (for CM's):
- RDHPCS
  - Intel
    - [x] Hera
    - [x] Orion
    - [x] Jet
    - [x] Gaea
    - [x] Cheyenne
  - GNU
    - [x] Hera
    - [x] Cheyenne
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- CI
  - [x] Completed
- opnReqTest
  - [ ] N/A
  - [x] Log attached to comment

### CODE REVIEW

**Comment by JessicaMeixner-NOAA** [comment]
No changes in gnu baseline, log: 
[RegressionTests_hera.gnu.log](https://github.com/ufs-community/ufs-weather-model/files/11337913/RegressionTests_hera.gnu.log)

Comparing against new baseline for intel, log: 
[RegressionTests_hera.intel.log](https://github.com/ufs-community/ufs-weather-model/files/11337915/RegressionTests_hera.intel.log)

**Comment by BrianCurtis-NOAA** [comment]
Will this change allow the test to be successful on wcoss2.intel ?

**Comment by JessicaMeixner-NOAA** [comment]
> Will this change allow the test to be successful on wcoss2.intel ?

Unsure, I will test and let you know.

**Comment by JessicaMeixner-NOAA** [comment]
@BrianCurtis-NOAA it is successful on WCOSS2 now.  I just ran the 2 32bit tests and they matched the baseline that I created.  I'm happy to share paths if needed.

**Comment by BrianCurtis-NOAA** [comment]
Thanks! Can you update rt.conf enabling wcoss2.intel for that test?

**Comment by JessicaMeixner-NOAA** [comment]
> Thanks! Can you update rt.conf enabling wcoss2.intel for that test?

@BrianCurtis-NOAA I think it's enabled by moving it to the other compile section already, but if I'm missing something here let me know, I'm happy to update this.

File: tests/rt.conf [hunk 1]
```diff
@@ -4,8 +4,8 @@
 
 COMPILE | -DAPP=S2SWA -D32BIT=ON -DCCPP_SUITES=FV3_GFS_v17_coupled_p8,FV3_GFS_cpld_rasmgshocnsstnoahmp_ugwp                       |                                         | fv3 |
 RUN     | cpld_control_p8_mixedmode                                                                                               |                                         | fv3 |
-COMPILE | -DAPP=S2SWA -DCCPP_SUITES=FV3_GFS_v17_coupled_p8,FV3_GFS_cpld_rasmgshocnsstnoahmp_ugwp                                  |                                         | fv3 |
 RUN     | cpld_control_gfsv17                                                                                                     | - wcoss2.intel                          | fv3 |
```
Comment (BrianCurtis-NOAA) [inline tests/rt.conf]: You'd need to remove the '- wcoss2.intel' here.

Comment (JessicaMeixner-NOAA) [reply tests/rt.conf]: I totally missed this.  Thanks for pointing it out @BrianCurtis-NOAA   I  also missed this when running my tests.  Let me re-check that this actually works before removing here.

**Comment by JessicaMeixner-NOAA** [comment]
@BrianCurtis-NOAA tested on wcoss2 and updated rt.conf

**Comment by jkbk2004** [comment]
@JessicaMeixner-NOAA We can start working on this pr. Can you sync up? #1701 contains a minor update on rt.conf. It will be great to combine #1701 into this pr. @DavidHuber-NOAA can you create a direct pr to Jessica's branch?

**Comment by JessicaMeixner-NOAA** [comment]
Branch is updated @jkbk2004

**Comment by zach1221** [comment]
Jenkins-ci logs attached, ORTs passed.
[ufs-weather-model » ort-docker-pipeline » PR-1728 #2 Console [Jenkins].pdf](https://github.com/ufs-community/ufs-weather-model/files/11489801/ufs-weather-model.ort-docker-pipeline.PR-1728.2.Console.Jenkins.pdf)

**Comment by BrianCurtis-NOAA** [comment]
WCOSS2 in maintenance mode. Acorn had a hardware failure on their ecflow node so i can't finish RT's, but baselines are done. Please skip both.

**Comment by jkbk2004** [comment]
Ok! We can start merging this pr. Please, go ahead for final reviews and approvals.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by zach1221 — APPROVED** [review]

**Review by FernandoAndrade-NOAA — APPROVED** [review]

**Pull request merged**